dd-trace-ot/src/test/groovy/datadog/opentracing/decorators/URLAsResourceNameTest.groovy

```diff
@@ -82,6 +82,7 @@ class URLAsResourceNameTest extends DDSpecification {
     "/V01/v9/abc/-1"                                   | "/V01/v9/abc/?"
     "/ABC/av-1/b_2/c.3/d4d/v5f/v699/7"                 | "/ABC/?/?/?/?/?/?/?"
     "/user/asdf123/repository/01234567-9ABC-DEF0-1234" | "/user/?/repository/?"
+    "/user/1/repo/test@example.com/"                   | "/user/?/repo/?/"
```

Comment:
Solely for the purposes of documenting behavior (not trying to get super-elaborate on "what is an email"), a few extra tests:
/user123@host-456.biz
/a@b@c
/@nothing-before-at.com
/endsinat@/includedsegment
/@

Again, I think it's totally fine to just let these all be elided but I feel it's important to clearly spell out the new behavior.